test/e2e/egress_test.go

```diff
@@ -335,34 +357,60 @@ func testEgressUpdateEgressIP(t *testing.T, data *TestData) {
 		originalNode     string
 		newNode          string
 		originalEgressIP string
+		oldIPRange       v1alpha2.IPRange
+		oldEgressIP      string
+		newIPRange       v1alpha2.IPRange
 		newEgressIP      string
 	}{
 		{
 			name:         "same Node",
 			originalNode: nodeName(0),
 			newNode:      nodeName(0),
+			oldIPRange:   v1alpha2.IPRange{CIDR: "169.254.100.0/30"},
+			oldEgressIP:  "169.254.100.1",
+			newIPRange:   v1alpha2.IPRange{CIDR: "169.254.101.0/30"},
+			newEgressIP:  "169.254.101.1",
 		},
 		{
 			name:         "different Nodes",
 			originalNode: nodeName(0),
 			newNode:      nodeName(1),
+			oldIPRange:   v1alpha2.IPRange{CIDR: "169.254.100.0/30"},
+			oldEgressIP:  "169.254.100.1",
+			newIPRange:   v1alpha2.IPRange{CIDR: "169.254.101.0/30"},
+			newEgressIP:  "169.254.101.1",
+		},
+		{
+			name:         "different Nodes in IPv6 cluster",
+			originalNode: nodeName(0),
+			newNode:      nodeName(1),
+			oldIPRange:   v1alpha2.IPRange{CIDR: "2021::aaa0/124"},
+			oldEgressIP:  "2021::aaa1",
+			newIPRange:   v1alpha2.IPRange{CIDR: "2021::bbb0/124"},
+			newEgressIP:  "2021::bbb1",
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			originalPool := data.createExternalIPPool(t, "originalpool-", v1alpha2.IPRange{CIDR: "169.254.100.0/30"}, nil, map[string]string{v1.LabelHostname: tt.originalNode})
+			if utilnet.IsIPv6String(tt.oldEgressIP) {
+				skipIfNotIPv6Cluster(t)
+			} else {
+				skipIfNotIPv4Cluster(t)
+			}
+			originalPool := data.createExternalIPPool(t, "originalpool-", tt.oldIPRange, nil, map[string]string{v1.LabelHostname: tt.originalNode})
 			defer data.crdClient.CrdV1alpha2().ExternalIPPools().Delete(context.TODO(), originalPool.Name, metav1.DeleteOptions{})
-			newPool := data.createExternalIPPool(t, "newpool-", v1alpha2.IPRange{CIDR: "169.254.101.0/30"}, nil, map[string]string{v1.LabelHostname: tt.newNode})
+			newPool := data.createExternalIPPool(t, "newpool-", tt.newIPRange, nil, map[string]string{v1.LabelHostname: tt.newNode})
 			defer data.crdClient.CrdV1alpha2().ExternalIPPools().Delete(context.TODO(), newPool.Name, metav1.DeleteOptions{})
-			originalIP := "169.254.100.1"
-			newIP := "169.254.101.1"
+			originalIP := tt.oldEgressIP
+			newIP := tt.newEgressIP
```

Comment:
nit: use `tt.oldEgressIP` and `tt.newEgressIP` directly. The local variables become pointless after this change.

Reply:
Done